### DISCUSSION TITLE
Cluster CouchDB version 2 and version 3

### DISCUSSION THREAD

**Discussion by H--o-l** [apache/couchdb#3042] [Q&A] [answered]
Hey!

I have two quick questions to the community.
Our team would like to migrate from CouchDB v2.3.1 to CouchDB v3.1.0.

We use a cluster of three machines, and to avoid downtime and do a smooth upgrade, we are thinking to upgrade machine by machine.
We use a one database-per-user model and thus have a lot of small dbs.

Our questions are, if we run a cluster made by **a mix of CouchDB v2 and CouchDB v3** machines:
- does data will be OK? (no corruption, no inconsistency?)
- does performance will be OK?

I tested on a (really small) test cluster and I read the [changelog](https://docs.couchdb.org/en/stable/whatsnew/3.0.html), all looks good, but before doing it in production I'd like to have feedback from you guys if you have to share.

PS: For example I'm aware that compaction could be an issue following https://github.com/apache/couchdb/issues/2941

**Comment by janl** [comment, marked as answer]
rolling upgrades like you describe are supported (modulo the bug you mentioned). We recommend to do those upgrades one after the other, so you don’t run in mixed mode too long. To make sure you don’t run into the bug, you could disable the replication scheduler and delete all .compact files, then do the upgrade (which enables the new replication schedule, so you don’t have to re-enable it). — be sure to adjust your `local.ini` as needed based on our release notes.

— 
Professional Support for Apache CouchDB:
https://neighbourhood.ie/couchdb-support/
